desertbot/modules/commands/Comic.py

```diff
@@ -0,0 +1,252 @@
+"""
+Created on Apr 18, 2019
+
+@author: lunik1
+"""
+import glob
+import json
+from io import BytesIO
+from random import sample
+
+import requests
+from PIL import Image, ImageDraw, ImageFont
+from twisted.plugin import IPlugin
+from zope.interface import implementer
+
+from desertbot.message import IRCMessage
+from desertbot.moduleinterface import IModule
+from desertbot.modules.commandinterface import BotCommand
+from desertbot.response import IRCResponse, ResponseType
+
+
+@implementer(IPlugin, IModule)
+class Comic(BotCommand):
+    # just hardcode a limit on max messages for now, should limit comic size pretty effecively?
+    messageLimit = 8
+
+    def actions(self):
+        # define additional actions so all PRIVMSGs and ACTIONs in the channel get handled by storeMessage
+        return super(Comic, self).actions() + [("message-channel", 1, self.storeMessage),
+                                               ("action-channel", 1, self.storeMessage)]
+
+    def onLoad(self) -> None:
+        # messages need to be stored for comics to work, store in a list as (nick, message)
+        # keys are channel names, as each channel kinda should have its own messages
+        self.messageStore = {}
+
+    def triggers(self):
+        return ['comic']
+
+    def help(self, query):
+        return 'comic - make a comic'
+
+    def execute(self, message: IRCMessage):
+        messages = self.getMessages(message.replyTo)
+        if len(messages) != 0:
+            comic = self.makeComic(messages)
+            return IRCResponse(ResponseType.Say, self.post_comic(comic), message.replyTo)
+
+    def getMessages(self, channel: str):
+        """
+        Get the messages for the given channel
+        Returns [] if there are no messages for that channel yet
+        """
+        return self.messageStore.get(channel, [])
+
+    def storeMessage(self, message: IRCMessage):
+        """
+        Store the message into the messageStore, and prune it to the limit if needed
+        """
+        # TODO just store the IRCMessage objects, so we can use other things than just the nick and message string
+        if message.command.lower() == "comic":
```

Comment:
Not sure if doing it this way is a good idea. Won't this check fail if we ever call this module from an alias?